src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Implementation/AppResourcesSI.cs

```diff
@@ -286,5 +286,76 @@ public string GetLayouts()
 
           return JsonConvert.SerializeObject(layouts);
         }
+
+        /// <inheritdoc />
+        public string GetLayoutSets()
+        {
+            string filename = Path.Join(_settings.AppBasePath, _settings.UiFolder, _settings.LayoutSetsFileName);
+            string filedata = null;
+            if (File.Exists(filename))
+            {
+                filedata = File.ReadAllText(filename, Encoding.UTF8);
+            }
+
+            return filedata;
+        }
+
+        /// <inheritdoc />
+        public string GetLayoutsForSet(string id)
+        {
+            Dictionary<string, object> layouts = new Dictionary<string, object>();
+
+            string layoutsPath = _settings.AppBasePath + _settings.UiFolder + id + "/layouts/";
+            if (Directory.Exists(layoutsPath))
+            {
+                foreach (string file in Directory.GetFiles(layoutsPath))
+                {
+                    string data = File.ReadAllText(file, Encoding.UTF8);
+                    string name = file.Replace(layoutsPath, string.Empty).Replace(".json", string.Empty);
+                    layouts.Add(name, JsonConvert.DeserializeObject<object>(data));
+                }
+            }
+
+            return JsonConvert.SerializeObject(layouts);
+        }
+
+        /// <inheritdoc />
+        public string GetLayoutSettingsForSet(string id)
+        {
+            string filename = Path.Join(_settings.AppBasePath, _settings.UiFolder, id, _settings.FormLayoutSettingsFileName);
+            string filedata = null;
+            if (File.Exists(filename))
+            {
+                filedata = File.ReadAllText(filename, Encoding.UTF8);
+            }
+
+            return filedata;
+        }
+
+        /// <inheritdoc />
+        public byte[] GetRuleConfigurationForSet(string id)
+        {
+            string filename = Path.Join(_settings.AppBasePath, _settings.UiFolder, id, _settings.RuleConfigurationJSONFileName);
+            byte[] filedata = new byte[0];
```

Comment:
This is initialized with a byte array, while the other methods are initialized with null. If this is this on purpose, write down the reason.

Reply:
Can the three methods with identical content use a common private method?

Reply:
Refactured